source/Calamari.Common/Plumbing/FileSystem/GlobExpressions/CharGroupRetriever.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Calamari.Common.Plumbing.FileSystem.GlobExpressions
+{
+    public static class CharGroupRetriever
```

Comment:
Is it worth having a set of tests _just_ for this specific logic? (rather than the overall result in `CalamariPhysicalFileSystemFixture`).

Reply:
Yep that's a fair call. I'll put some in.